tests/copy-paste.js

```diff
@@ -0,0 +1,780 @@
+            ], ["Cannot insert Choice into or after Text"]);
+            util.selectAll();
+            eq(mod.copy(), [
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/text", "Text", "text", "text"],
+            ]);
+        });
+
+        it("should not overwrite itext with auto-id", function () {
+            util.loadXML("");
+            paste([
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/text", "Text", "auto", "auto"],
+            ]);
+            util.clickQuestion("text");
+            paste([
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default", "labelItext"],
+                ["/text2", "Text", "non", "non", "text-label"],
+            ]);
+            util.selectAll();
+            eq(mod.copy(), [
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default", "labelItext"],
+                ["/text", "Text", "auto", "auto", "null"],
+                ["/text2", "Text", "non", "non", "text-label"],
+            ]);
+            call("createXML");
+            var text = util.getMug("text"),
+                text2 = util.getMug("text2");
+            // NOTE which one gets auto-renamed is somewhat arbitrary
+            assert.equal(text.p.labelItext.autoId, true);
+            assert.equal(text.p.labelItext.id, "text-label");
+            assert.equal(text2.p.labelItext.autoId, false);
+            assert.equal(text2.p.labelItext.id, "text-label2");
+        });
+
+        it("should fill empty itext forms", function () {
+            util.loadXML("");
+            paste([
+                ["id", "type", "labelItext:en-default"],
+                ["/text1", "Text", "Label"],
+                ["/text2", "Text", ""],
+                ["/text3", "Text", "null"],
+                ["/text4", "Text"],
+            ]);
+            util.selectAll();
+            eq(mod.copy(), [
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/text1", "Text", "Label", "Label"],
+                ["/text2", "Text", "text2", "text2"],
+                ["/text3", "Text", "text3", "text3"],
+                ["/text4", "Text", "text4", "text4"],
+            ]);
+        });
+
+        it("should fill empty default itext forms", function () {
+            util.loadXML("");
+            paste([
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/text", "Text", "", "Label"],
+            ]);
+            util.selectAll();
+            eq(mod.copy(), [
+                ["id", "type", "labelItext:en-default", "labelItext:hin-default"],
+                ["/text", "Text", "Label", "Label"],
+            ]);
+        });
+
+        it("should ignore extra cells in header row", function () {
+            util.loadXML("");
+            paste(tsv.tabDelimit([
+                HEADER.concat(["", ""]),
```

Comment:
I'm sure it does, given how it's [comparing the preamble](https://github.com/dimagi/Vellum/pull/380/files#diff-815d962c2df9702e02af486ea3984d4bR251). The extra cells are ignored. Do you think that's a problem?